community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/helpers/JavaConversionSupport.scala

```diff
@@ -42,4 +41,33 @@ object JavaConversionSupport
     def hasNext = iterator.hasNext
     def next() = f(iterator.next())
   }
+
+  // Same as mapToScala, but handles concurrency exceptions by swallowing exceptions
+  def mapToScalaENFXSafe[T](iterator: PrimitiveLongIterator)(f: Long => T): Iterator[T] = new Iterator[T] {
+    private var _next: Option[T] = fetchNext()
```

Comment:
I'm curious: when is this evaluated? At the first read of `_next` or when the object is initialized?